### PR TITLE
fix(runtime-fuzzer): update wasm-smith fork & disable all traps in wasm

### LINKED ISSUE #3382: Lower amount of traps when `Control` instructions are off for wasms from `wasm-gen`
### Proposal

If you turn off `Control` instructions from `wasm-gen`, the execution will become really fast, but the rate of `Traps` (`Possibly 'unreachable' instruction occurred`) becomes too high. There are basically 2 main reasons (wasmi): `out of bounds memory access` and `integer divide by zero`. The task is to lower the rate of these erros.

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Resolves #3382

TODO:
- [x] make sure that the new version of wasm-smith does not generate WASMs that may be rejected on our pallet by checking logs
      run fuzzer with `RUST_LOG="pallet_gear=debug"` and `| grep "Code failed to load"`
- [x] merge PR in our wasm-tools repo: https://github.com/gear-tech/wasm-tools/pull/4 and use `gear-stable` branch in `Cargo.toml`
  diff between original wasm-smith: https://github.com/bytecodealliance/wasm-tools/compare/main...gear-tech:wasm-tools:av/merge-upstream
  TL;DR: they added `disallow_traps` option, our options `call_indirect_enabled` and `unreachable_enabled` is obsolete
- [x] uncomment `Control` in this PR

### CODE REVIEW

**Review by mertwole — APPROVED** [review]

**Comment by breathx** [comment]
Why is it in pleasereview status if still none of all todos from your own list in description solved?

**Review by breathx — APPROVED** [review]
@SabaunT please take a look

**Review by SabaunT — APPROVED** [review]
Great work and findings.
I have run fuzzer from this PR 1) with/without control instruction and 2) compared that to master using 3) the same starting seed and corpuses set.

I'm not sure whether it really helped a lot to increase amount of successful executions. Maybe not significantly. But it changed a lot the structure traps, i.e. there are no more `out of bounds memory access`, which was the main error from `wasmi`. However, there are a lot of `wasm 'unreachable' instruction...` errors.

Must be stated that the PR doesn't affect `Not enough gas to continue execution`.

Turning off control instructions 1) brings huge speed (in around 4-5 minutes fuzzer stops with oom error), 2) drastically lowers amount of `Not enough gas...` errors. However, it brings lots of `Sys-call unrecoverable error` traps.
